src/RestaurantApi/Users/AddUser.js

```diff
@@ -0,0 +1,40 @@
+import { auth, db } from "../../firebase"
+import { collection, addDoc} from 'firebase/firestore'
+import { createUserWithEmailAndPassword } from "firebase/auth";
+
+
+const addUserInfo = async (email, password, name, phoneNumber, accountType="user", id) => {
+  const usersCollectionRef = collection(db, 'users');
+
+  try {
+    await addDoc(usersCollectionRef, {
+      user_id: id, 
+      email: email, 
+      name: name, 
+      phone_number: phoneNumber, 
+      account_type: accountType,
+    });
+    
+  } catch (e) {
+    console.error(e.message, e.name);
+  }
+};
+
+const addUser = async (email, password, name, phoneNumber, accountType) => {
+  createUserWithEmailAndPassword(auth, email, password)
+  .then((userCredential) => {
+    const user = userCredential.user;
+    addUserInfo(email, password, name, phoneNumber, accountType, user.uid)
+    .then(() => {
+      console.log("The user was added");
+    })
+    .catch((e) => {
+      console.error(e.message, e.name);
+    });
+  })
+  .catch((e) => {
```

Comment:
hmm you can do like this:

```
createUserWithEmailAndPassword(...)
   .then( (userCredntial) => {
      return userCredential.user.uid
   })
   .then( (uid) => {
      return addUserInfo(...)
   })
   .then( (response) => {
      return reponse.json // or something that you need in your response
   })
   .catch( (err) => {
      throw 'Server error or something'
   })
```
This _.catch_ on the end will catch anything wrong that would happen on any _.then_

It is better to keep async code linear rather than nested, if you nest some .then you will have to nesting callbacks and that is a "highway" to callback hell. You can try above code or just make async function whit await statements before each async method for example:
`const userCredentials = await createUserWithEmailAndPassword(auth, email, password)` 
and after that statemenbt you can use uderCredentials variable like in normal code. But be aware that you have to use try ... catch to catch some errors that can occur so:

```
try {
  const userCredentials = await createUserWithEmailAndPassword(auth, email, password)
  /... some code here
}
  catch(err) {
  throw 'error'
}
```